Expensify/App
# Improve attachment validation on the front-end

Adds validation for minimum file size and file type.

### Fixed Issues
$ https://github.com/Expensify/App/issues/10050

### Tests
1. Upload a file that is too small (less than 240 bytes). Here is one you can use:
![icon16](https://user-images.githubusercontent.com/1228807/181106778-e075ee9a-7918-48e5-a6dc-dc4d8f4bea16.png)
1. Verify that you get a too small error message
2. Upload an image that is too larger (more than 50 MB). You can download this high-res image from here: https://effigis.com/wp-content/themes/effigis_2014/img/RapidEye_RapidEye_5m_RGB_Altotting_Germany_Agriculture_and_Forestry_2009MAY17_8bits_sub_r_2.jpg
4. Verify that you get a too big error message
5. Make a copy of that small file from above and change it's extension to something invalid (like `.x`) and upload it
6. Verify that you get a wrong file type error message
7. Upload proper files and verify they upload properly


### QA Steps
Same as tests


### Screenshots


#### Web
<img width="391" alt="2022-07-26_14-55-51" src="https://user-images.githubusercontent.com/1228807/181110386-d4349dfa-47f1-4422-abd9-f9cb0e84f25e.png">
<img width="397" alt="2022-07-26_14-55-37" src="https://user-images.githubusercontent.com/1228807/181110397-7ddc488c-54a0-4add-9688-5c49b0c4db8f.png">
<img width="415" alt="2022-07-26_14-53-07" src="https://user-images.githubusercontent.com/1228807/181110405-6616dc81-ac14-40b9-ab16-31011c1130c4.png">


#### Mobile Web
<img width="370" alt="2022-07-26_14-58-14" src="https://user-images.githubusercontent.com/1228807/181110694-f8b844ac-a1ad-4cd9-88f4-8fbdbdd100e9.png">
<img width="370" alt="2022-07-26_14-58-02" src="https://user-images.githubusercontent.com/1228807/181110713-2e280989-f7ee-4d3e-9561-958ccc01aa93.png">
<img width="372" alt="2022-07-26_14-58-30" src="https://user-images.githubusercontent.com/1228807/181110718-ec1ffcc2-028d-4006-b637-f3cfeb27e9ea.png">


#### Desktop
<img width="397" alt="2022-07-26_15-01-26" src="https://user-images.githubusercontent.com/1228807/181111327-91b111de-47b2-467d-b086-68355bd366d9.png">
<img width="397" alt="2022-07-26_15-01-38" src="https://user-images.githubusercontent.com/1228807/181111339-a86045f5-8dd5-480a-981c-b3eb8df9e616.png">
<img width="394" alt="2022-07-26_15-01-51" src="https://user-images.githubusercontent.com/1228807/181111349-84173ebc-15eb-4ffa-b242-ed700dcc6f0a.png">


#### iOS
I'm struggling to get iOS running ([see slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1658870502226249?thread_ts=1656097944.067329&cid=C01GTK53T8Q))

#### Android
I couldn't really find any way to test the errors because it's difficult to force those kind of size images on the emulator. I tried downloading some images from the web, but I still had trouble finding them to upload into the app.
